core/src/test/java/org/wso2/carbon/kernel/utils/UtilsTest.java

```diff
@@ -167,4 +167,42 @@ public void testGetCarbonHome() throws Exception {
         }
         setEnvironmentalVariables(backup);
     }
+
+    @Test
+    public void testSubstituteVarsSystemPropertyNotNull() {
+        String carbonHome = System.getProperty(Constants.CARBON_HOME);
+        Boolean needToClearCarbonHomeAtTheEnd = false;
+
+        if (carbonHome == null) {
+            carbonHome = "test-carbon-home";
+            System.setProperty(Constants.CARBON_HOME, carbonHome);
+            needToClearCarbonHomeAtTheEnd = true;
+        }
+
+        Assert.assertEquals(Utils.substituteVars("${carbon.home}"), carbonHome);
+
+        if (needToClearCarbonHomeAtTheEnd) {
```

Comment:
Use TestNg annotation for this If you do this after class it will make sure it clears up after each test execution , Here if assert is false then property will not be cleared